packages/site-parsers/src/parsers/wix/mappers.js

```diff
@@ -52,22 +63,21 @@ module.exports = {
 	},
 
 	componentMapper: ( component, meta ) => {
-		if ( ! component.dataQuery ) {
```

Comment:
Double checking on this logic change, it seems like removing this check causes two potential issues:

- Subsequent usage of `component.dataQuery` in `componentMapper()` requires it to be checked that it's not undefined.
- As `wrapResult()` adds the (now sometimes undefined) `designQuery` to the block data, it could cause flow on issues elsewhere. I _think_ `resolveQueries()` is safe, for example, but it's good to be sure.

Is it possible to make this safer?

Reply:
Actually, we do check the `dataQuery.type` not to be an undefined value.
Please see [line 70](https://github.com/pento/free-as-in-speech/pull/60/files#diff-fe948eb4b0b12478cb0fc8d776eadf67693a1323ca00ddc6e7ad44871e6a9083R70) and [line 77](https://github.com/pento/free-as-in-speech/pull/60/files#diff-fe948eb4b0b12478cb0fc8d776eadf67693a1323ca00ddc6e7ad44871e6a9083R77).

To clarify, with these changes, we are allowing mappers to match type by field `type` or the last segment of `componentType` value string.

Reply:
Thanks for clarifying. 🙂